### PR TITLE
Calculation of proposing costs for blob and calldata

### PR DESCRIPTION
**Pull request by mskrzypkows**

### CODE REVIEW

File: packages/taiko-client/proposer/proposer.go [hunk 1]
```diff
@@ -537,40 +668,84 @@ func (p *Proposer) Name() string {
...
-		if err != nil {
-			return false, err
+	baseL2Fee, err := p.rpc.L2.SuggestGasPrice(p.ctx)
+	if err != nil {
+		return nil, err
+	}
+	priorityL2Fee, err := p.rpc.L2.SuggestGasTipCap(p.ctx)
+	if err != nil {
+		return nil, err
+	}
```
Comment (smartprogrammer93) [inline packages/taiko-client/proposer/proposer.go:698]: These should come from the l2 tx not from the L2 rpc

**Review by smartprogrammer93 — CHANGES_REQUESTED** [review]

File: packages/taiko-client/proposer/proposer.go (hunk 1 above)
Comment (mskrzypkows) [reply packages/taiko-client/proposer/proposer.go]: Fixed

File: packages/taiko-client/driver/driver_test.go [hunk 2]
```diff
@@ -340,10 +340,10 @@ func (s *DriverTestSuite) InitProposer() {
 			TxSendTimeout:             txmgr.DefaultBatcherFlagValues.TxSendTimeout,
 			TxNotInMempoolTimeout:     txmgr.DefaultBatcherFlagValues.TxNotInMempoolTimeout,
 		},
-		GasNeededForProposingBlock: 0,
-		GasNeededForProvingBlock:   0,
-		PriceFluctuationModifier:   50,
-		OffChainCosts:              big.NewInt(0),
+		CheckProfitability:       false,
+		GasNeededForProvingBlock: 0,
+		PriceFluctuationModifier: 50,
```
Comment (smartprogrammer93) [inline packages/taiko-client/driver/driver_test.go:345]: set this to zero in tests

File: packages/taiko-client/driver/chain_syncer/chain_syncer_test.go [hunk 3]
```diff
@@ -100,10 +100,10 @@ func (s *ChainSyncerTestSuite) SetupTest() {
 			TxSendTimeout:             txmgr.DefaultBatcherFlagValues.TxSendTimeout,
 			TxNotInMempoolTimeout:     txmgr.DefaultBatcherFlagValues.TxNotInMempoolTimeout,
 		},
-		GasNeededForProposingBlock: 0,
-		GasNeededForProvingBlock:   0,
-		PriceFluctuationModifier:   50,
-		OffChainCosts:              big.NewInt(0),
+		CheckProfitability:       false,
+		GasNeededForProvingBlock: 0,
+		PriceFluctuationModifier: 50,
```
Comment (smartprogrammer93) [inline packages/taiko-client/driver/chain_syncer/chain_syncer_test.go:105]: set to zero

File: packages/taiko-client/driver/chain_syncer/blob/syncer_test.go [hunk 4]
```diff
@@ -250,10 +250,10 @@ func (s *BlobSyncerTestSuite) initProposer() {
 			TxSendTimeout:             txmgr.DefaultBatcherFlagValues.TxSendTimeout,
 			TxNotInMempoolTimeout:     txmgr.DefaultBatcherFlagValues.TxNotInMempoolTimeout,
 		},
-		GasNeededForProposingBlock: 0,
-		GasNeededForProvingBlock:   0,
-		PriceFluctuationModifier:   50,
-		OffChainCosts:              big.NewInt(0),
+		CheckProfitability:       false,
+		GasNeededForProvingBlock: 0,
+		PriceFluctuationModifier: 50,
```
Comment (smartprogrammer93) [inline packages/taiko-client/driver/chain_syncer/blob/syncer_test.go:255]: set to zero

File: packages/taiko-client/cmd/flags/proposer.go [hunk 5]
```diff
@@ -112,33 +112,31 @@ var (
...
 		Value:    0,
 		Category: proposerCategory,
 		EnvVars:  []string{"SURGE_GAS_NEEDED_FOR_PROVING_BLOCK"},
 	}
 
 	PriceFluctuationModifier = &cli.Uint64Flag{
-		Name:     "surge.priceFluctuationModifier",
+		Name:     "priceFluctuationModifier",
 		Usage:    "Price fluctuation modifier in percentage",
 		Value:    50,
```
Comment (smartprogrammer93) [inline packages/taiko-client/cmd/flags/proposer.go:133]: set to zero by default

File: packages/taiko-client/proposer/proposer_test.go [hunk 6]
```diff
@@ -113,10 +113,10 @@ func (s *ProposerTestSuite) SetupTest() {
 			TxSendTimeout:             txmgr.DefaultBatcherFlagValues.TxSendTimeout,
 			TxNotInMempoolTimeout:     txmgr.DefaultBatcherFlagValues.TxNotInMempoolTimeout,
 		},
-		GasNeededForProposingBlock: 0,
-		GasNeededForProvingBlock:   0,
-		PriceFluctuationModifier:   50,
-		OffChainCosts:              big.NewInt(0),
+		CheckProfitability:       true,
```
Comment (smartprogrammer93) [inline packages/taiko-client/proposer/proposer_test.go:116]: why enabling this for all tests, should only be done on tests special to the profitability part

File: packages/taiko-client/proposer/proposer_test.go [hunk 7]
```diff
@@ -113,10 +113,10 @@ func (s *ProposerTestSuite) SetupTest() {
...
+		GasNeededForProvingBlock: 0,
+		PriceFluctuationModifier: 50,
+		OffChainCosts:            big.NewInt(0),
```
Comment (smartprogrammer93) [inline packages/taiko-client/proposer/proposer_test.go:119]: I think all these values should be left on defualt unless you are explicitly testing them

File: packages/taiko-client/proposer/proposer.go [hunk 8]
```diff
@@ -309,28 +303,28 @@ func (p *Proposer) ProposeOp(ctx context.Context) error {
...
 
-// ProposeTxList proposes the given transactions lists to TaikoL1 smart contract.
+// ProposeTxLists proposes the given transactions lists to TaikoL1 smart contract.
 func (p *Proposer) ProposeTxLists(ctx context.Context, txLists []types.Transactions) error {
 	// Check if the current L2 chain is after ontake fork.
 	state, err := rpc.GetProtocolStateVariables(p.rpc.TaikoL1, &bind.CallOpts{Context: ctx})
 	if err != nil {
 		return err
 	}
+	isOntake := p.chainConfig.IsOntake(new(big.Int).SetUint64(state.B.NumBlocks))
```
Comment (smartprogrammer93) [inline packages/taiko-client/proposer/proposer.go:322]: no need for fork specific code in proposer. treat everything as ontake

**Review by smartprogrammer93 — CHANGES_REQUESTED** [review]
Lets also remove all ontake references from here. even in logs.

**Review by jeffrydegrande — APPROVED** [review]

**Review by smartprogrammer93 — APPROVED** [review]

**Pull request merged**